platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/StatsToOpenMetricCommand.java

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.swirlds.platform.cli;
+
+import com.swirlds.cli.PlatformCli;
+import com.swirlds.cli.utility.AbstractCommand;
+import com.swirlds.cli.utility.SubcommandOf;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+        name = "stats-to-open-metrics",
+        mixinStandardHelpOptions = true,
+        description = "Converts list of stat CSV files to OpenMetric format")
+@SubcommandOf(PlatformCli.class)
+public class StatsToOpenMetricCommand extends AbstractCommand {
+
+    private List<Path> csvFiles;
+    private BufferedReader lineReader;
+    private String currentLine;
+    private String[] firstHeaders;
+    private String[] secondHeaders;
+    private int timeIndex;
+    private String sampleName;
+    private Path output;
+    private BufferedWriter outputFile;
+
+    @CommandLine.Parameters(description = "The csv stat files to read")
+    private void setTestData(@NonNull final List<Path> csvFiles) {
+        this.csvFiles = csvFiles;
+    }
+
+    @CommandLine.Option(
+            names = {"-o", "--output-file"},
+            description = "Path to output file for resulting metrics")
+    private void setFile(@NonNull final Path file) {
+        this.output = file;
+    }
+
+    @Override
+    public Integer call() throws Exception {
+
+        this.outputFile = new BufferedWriter(new FileWriter(this.output.toFile()));
+        for (final Path csvFile : this.csvFiles) {
+            System.out.println("Processing stat file: " + csvFile);
+            openFile(csvFile);
+            try {
+                skipUselessHeaders();
+                readRealHeaders();
+                processDataLines();
+            } finally {
+                this.lineReader.close();
```

Comment:
If an exception is thrown mid-loop, outputFile may not be closed, and lineReader.close() is only in a finally block for each CSV, but the writer is not.